backend/app/usecase/search/search.go

```diff
@@ -66,13 +71,85 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		s.logger.Error(errors.New("user not provided"))
+		return Result{}, nil
+	}
+
+	shortLinks, err := s.getShortLinkByUser(*query.User)
+	if err != nil {
+		return Result{}, err
+	}
+
+	var matchedAliasByAll, matchedAliasByAny, matchedLongLinkByAll, matchedLongLinkByAny []entity.ShortLink
+	keywords := getKeywords(query.Query, " ")
+	for _, shortLink := range shortLinks {
+		if matcher.ContainsAll(keywords, shortLink.Alias) {
+			matchedAliasByAll = append(matchedAliasByAll, shortLink)
+		} else if matcher.ContainsAny(keywords, shortLink.Alias) {
+			matchedAliasByAny = append(matchedAliasByAny, shortLink)
+		} else if matcher.ContainsAll(keywords, shortLink.LongLink) {
+			matchedLongLinkByAll = append(matchedLongLinkByAll, shortLink)
+		} else if matcher.ContainsAny(keywords, shortLink.LongLink) {
+			matchedLongLinkByAny = append(matchedLongLinkByAny, shortLink)
+		}
+	}
+
+	sortShortLinks(orderBy, matchedAliasByAll, matchedAliasByAny, matchedLongLinkByAll, matchedLongLinkByAny)
+
+	mergedShortLinks := mergeShortLinks(
+		matchedAliasByAll,
+		matchedAliasByAny,
+		matchedLongLinkByAll,
+		matchedLongLinkByAny,
+	)
+
+	filteredShortLinks := filterShortLinks(mergedShortLinks, filter)
+
+	return Result{
+		shortLinks: filteredShortLinks,
+		users:      nil,
+	}, nil
 }
 
 func (s Search) searchUser(query Query, orderBy order.Order, filter Filter) (Result, error) {
 	return Result{}, nil
 }
 
+func (s Search) getShortLinkByUser(user entity.User) ([]entity.ShortLink, error) {
+	aliases, err := s.userShortLinkRepo.FindAliasesByUser(user)
+	if err != nil {
+		return []entity.ShortLink{}, err
+	}
+
+	return s.shortLinkRepo.GetShortLinksByAliases(aliases)
+}
+
+func getKeywords(query, separator string) []string {
```

Comment:
```suggestion
func getKeywords(query string) []string {
Doing split is an implementation detail of obtaining the keywords from the query. This can change in the future.
```